File: backend/app/schemas/misc_sale.py
import uuid
from datetime import datetime
from decimal import Decimal

from sqlmodel import SQLModel

from app.models import MiscSaleBase


class MiscSalePublic(MiscSaleBase):
    id: uuid.UUID
    account_name: str
    
class MiscSaleCreate(MiscSaleBase):
    account_id: uuid.UUID
    
    
class MiscSaleUpdate(SQLModel):
    amount: Decimal | None = None
    date: datetime | None = None
    reference_number: str | None = None
    account_id: uuid.UUID | None = None
    description: str | None = None

File: frontend/src/types/miscSale.ts
interface MiscSaleBase {
    date: string
    description: string
    reference_number?: string
    account_id: string
}

export interface MiscSale extends MiscSaleBase {
    id: string
    amount: string
    account_name: string
}

export interface MiscSaleCreate extends MiscSaleBase {
    amount: number
}

export interface MiscSaleUpdate {
    date?: string
    description?: string
    reference_number?: string | null
    amount?: number
    account_id?: string
}